src/HttpServer/HttpServer.cc

```diff
@@ -56,16 +61,28 @@ void HttpServer::RegisterRoutes() {
     }
 
     if (_enable_database) {
-        // Dynamic routes for preferences, layouts and snippets
+        // Dynamic routes for preferences, layouts, snippets and workspaces
         app.get("/api/database/preferences", [&](auto res, auto req) { HandleGetPreferences(res, req); });
         app.put("/api/database/preferences", [&](auto res, auto req) { HandleSetPreferences(res, req); });
         app.del("/api/database/preferences", [&](auto res, auto req) { HandleClearPreferences(res, req); });
+
+        app.get("/api/database/list/layouts", [&](auto res, auto req) { HandleGetObjectList("layout", res, req); });
         app.get("/api/database/layouts", [&](auto res, auto req) { HandleGetObjects("layout", res, req); });
+        app.get("/api/database/layout/:name", [&](auto res, auto req) { HandleGetObject("layout", res, req); });
         app.put("/api/database/layout", [&](auto res, auto req) { HandleSetObject("layout", res, req); });
         app.del("/api/database/layout", [&](auto res, auto req) { HandleClearObject("layout", res, req); });
+
+        app.get("/api/database/list/snippets", [&](auto res, auto req) { HandleGetObjectList("snippet", res, req); });
         app.get("/api/database/snippets", [&](auto res, auto req) { HandleGetObjects("snippet", res, req); });
+        app.get("/api/database/snippet/:name", [&](auto res, auto req) { HandleGetObject("snippet", res, req); });
         app.put("/api/database/snippet", [&](auto res, auto req) { HandleSetObject("snippet", res, req); });
         app.del("/api/database/snippet", [&](auto res, auto req) { HandleClearObject("snippet", res, req); });
+
+        app.get("/api/database/list/workspaces", [&](auto res, auto req) { HandleGetObjectList("workspace", res, req); });
+        app.get("/api/database/workspaces", [&](auto res, auto req) { HandleGetObjects("workspace", res, req); });
+        app.get("/api/database/workspace/:name", [&](auto res, auto req) { HandleGetObject("workspace", res, req); });
+        app.put("/api/database/workspace", [&](auto res, auto req) { HandleSetObject("workspace", res, req); });
+        app.del("/api/database/workspace", [&](auto res, auto req) { HandleClearObject("workspace", res, req); });
```

Comment:
Now that I'm looking at this again, I don't love the use of magic hardcoded strings here, and would prefer it if we created constants for these (or maybe an enum would be cleaner). But this is a minor issue which predates this PR, so we could fix it later.

Reply:
can I suggest we make this and the suggestion below a separate issue that we solve after code freeze?

Reply:
Works for me; I'd rather not mess around with a working PR now. :)